foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/PythonLintingMigration.java

```diff
@@ -0,0 +1,153 @@
+        Object configuration;
+
+        if (habushuPlugin == null) {
+            return true;
+        } else {
+            configuration = getConfiguration(habushuPlugin);
+        }
+        
+        if (configuration == null) {
+            return true;
+        }
+        
+        if (!hasLinting(configuration)) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin habushuPlugin = getHabushuPlugin(model);
+        Object configuration;
+        if (habushuPlugin != null) {
+            configuration = getConfiguration(habushuPlugin);
+        } else {
+            configuration = null;
+        }
+        int indent;
+        InputLocation inputLocation;
+        String line;
+        ArrayList<String> lines;
+        
+        if (habushuPlugin == null) {
+            InputLocation referenceLocation = model.getBuild().getPluginManagement().getLocation(LocationAwareMavenReader.END);
+            inputLocation = new InputLocation(referenceLocation.getLineNumber() - 1, referenceLocation.getColumnNumber());
+            lines = new ArrayList(Arrays.asList(
+                "<plugin>",
+                "<groupId>org.technologybrewery.habushu</groupId>",
+                "<artifactId>habushu-maven-plugin</artifactId>",
+                "<configuration>",
+                "<sourceFailOnLintErrors>false</sourceFailOnLintErrors>",
+                "<testFailOnLintErrors>false</testFailOnLintErrors>",
+                "</configuration>",
+                "</plugin>"
+            ));
+            indent = 4;
+        } else if (configuration == null) {
+            InputLocation referenceLocation = habushuPlugin.getLocation(LocationAwareMavenReader.END);
+            inputLocation = new InputLocation(referenceLocation.getLineNumber(), referenceLocation.getColumnNumber());
+            lines = new ArrayList(Arrays.asList(
+                "<configuration>",
+                "<sourceFailOnLintErrors>false</sourceFailOnLintErrors>",
+                "<testFailOnLintErrors>false</testFailOnLintErrors>",
+                "</configuration>"
+            ));
+            indent = 5;
+        } else {
+            InputLocation referenceLocation = habushuPlugin.getLocation(LocationAwareMavenReader.END);
+            inputLocation = new InputLocation(referenceLocation.getLineNumber() - 1, referenceLocation.getColumnNumber());
+            lines = new ArrayList(Arrays.asList(
+                "<sourceFailOnLintErrors>false</sourceFailOnLintErrors>",
+                "<testFailOnLintErrors>false</testFailOnLintErrors>"
+            ));
+            indent = 6;
+        }
+        line = configureLine(lines, indent);
+        PomModifications modifications = new PomModifications();
+        modifications.add(new PomModifications.Insertion(inputLocation, indent, l -> line));
```

Comment:
S: Just use the indent from the modifications framework:
```suggestion
        modifications.add(new PomModifications.Insertion(inputLocation, indent, indentStr -> {
            lines.stream()
                    .map(line -> StringUtils.repeat(indentStr, indent) + line)
                    .collect(Collectors.joining("\n"))
        }));
```

This is just an example. To get the relative indents, best to pull the lines out into a method and just increment as appropriate. Something like:
```java
private static String getHabushuPlugin(String indent) {
    return StringUtils.repeat(indent, 4) + "<plugin>\n" +
               StringUtils.repeat(indent, 5) +  "<groupId>org.technologybrewery.habushu</groupId>\n" + 
               ... + getHabushuConfiguration(indent) + "\n" +
               StringUtils.repeat(indent, 4) "</plugin>";
```